src/core/lib/gpr/sync_posix.cc

```diff
@@ -72,27 +73,58 @@ gpr_atm gpr_counter_atm_add = 0;
 #endif
 
 void gpr_mu_init(gpr_mu* mu) {
+#ifdef GRPC_ASAN_ENABLED
+  GPR_ASSERT(pthread_mutex_init(&mu->mutex, nullptr) == 0);
+  mu->leak_checker = (int*)gpr_malloc(sizeof(*mu->leak_checker));
+  GPR_ASSERT(mu->leak_checker != nullptr);
+  /* Initial it with a magic number, make no sense, just use the memory.
```

Comment:
Could you explain to me why we need to initialize the memory with a magic number? Is just allocating it not enough?

Reply:
It seems this is unnecessary. I just wanted to do an initialization, forgot that we just need the leak detection. I'll remove it later.

Reply:
done